pkg/datapath/cells.go

```diff
@@ -172,77 +180,26 @@ func newWireguardAgent(lc cell.Lifecycle, sysctl sysctl.Sysctl) *wg.Agent {
 	return wgAgent
 }
 
-func newDatapath(params datapathParams) types.Datapath {
-	datapath := linuxdatapath.NewDatapath(linuxdatapath.DatapathParams{
-		ConfigWriter:   params.ConfigWriter,
-		RuleManager:    params.IptablesManager,
-		WGAgent:        params.WgAgent,
-		NodeMap:        params.NodeMap,
-		NodeAddressing: params.NodeAddressing,
-		BWManager:      params.BandwidthManager,
-		NodeManager:    params.NodeManager,
-		DB:             params.DB,
-		Devices:        params.Devices,
-		Orchestrator:   params.Orchestrator,
-		NodeHandler:    params.NodeHandler,
-		NodeNeighbors:  params.NodeNeighbors,
-		ExpConfig:      params.ExpConfig,
-	})
-
+func initDatapath(params datapathParams) {
 	params.LC.Append(cell.Hook{
 		OnStart: func(cell.HookContext) error {
 			if err := linuxdatapath.CheckRequirements(params.Log); err != nil {
 				return fmt.Errorf("requirements failed: %w", err)
 			}
 
-			params.NodeIDHandler.RestoreNodeIDs()
 			return nil
 		},
 	})
-
-	return datapath
 }
 
 type datapathParams struct {
 	cell.In
 
 	Log *slog.Logger
 
-	LC      cell.Lifecycle
-	WgAgent *wg.Agent
+	LC cell.Lifecycle
 
 	// Force map initialisation before loader. You should not use these otherwise.
```

Comment:
should we move this dependency to the loader cell?

Reply:
Yes it should definitely move there!